### ISSUE TITLE
Vet not able to run on PR.

### ISSUE THREAD

**Issue by shubham-chaskar** [safedep/vet-action#66]
Hello Team, We have recently faced a new issue on PR. We are not able to find root cause and can also not provide much information as it is not avaliable in logs too. 

**Below is the screenshot of the error.**

![Screenshot 2024-09-18 at 1 20 54 PM](https://github.com/user-attachments/assets/b703acc9-cf13-4bbc-aa94-c1e594fce158)

Below is the commit history. The later one has the failed github action.(af03918)

![Screenshot 2024-09-18 at 1 21 37 PM](https://github.com/user-attachments/assets/b9e20d95-b13a-4b2a-88ff-c38d2781aa8f)

Thank you. Please let me know if any more information is needed from our end.

**Comment by abhisek** [comment]
@shubham-chaskar Thanks for reporting this. Can you confirm if the PR source (HEAD) branch is from the same repository or from a different (forked) repository?

**Comment by shubham-chaskar** [comment]
Hello @abhisek it is from the same repository.

**Comment by abhisek** [comment]
@shubham-chaskar I am guess the one failed has something to do with "merge" commit. Was a PR from `master` branch to `DEVOPS-4203`? It will be great to know the source (head) and destination (base) branch used in the PR where `vet` failed.

**Comment by shubham-chaskar** [comment]
@abhisek Here are some more details,

<img width="893" alt="Screenshot 2024-09-19 at 11 43 57 AM" src="https://github.com/user-attachments/assets/899ef389-1bee-4560-9162-dd471180669b">

<img width="970" alt="Screenshot 2024-09-19 at 11 41 39 AM" src="https://github.com/user-attachments/assets/00e43a54-b293-4805-8f5e-7ecce2e142b7">


The PR was from devops-4203 to master branch, in the same PR later master was merged to devops-4203 which triggers vet failure.

**Comment by abhisek** [comment]
@shubham-chaskar I briefly looked at the issue. It seems to me merging base branch into head branch caused an issue with commits getting out of order. It may be possible that base is ahead of head. Usually it is a good idea to do `git rebase` to keep things linear. I will see if we can handle this case but seems like need some handling on top of GitHub API.

**Comment by shubham-chaskar** [comment]
@abhisek Whenever base is ahead of head we get the below error, is this what you are referring to?

```
 Error: Head is not ahead of Base: diverged
```

**Comment by abhisek** [comment]
@shubham-chaskar Yes. You are right. The error is usually different when head is behind base. Still looking at this.

**Comment by abhisek** [comment]
@shubham-chaskar We have made some improvements in how this action computes the changes introduced in this PR. While I am still unable to check for your specific case, I think it should be handled now.
